core/include/traccc/definitions/concepts.hpp

```diff
@@ -0,0 +1,15 @@
+/**
+ * traccc library, part of the ACTS project (R&D line)
+ *
+ * (c) 2024 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+#if __cpp_concepts >= 201907L
+#define CONSTRAINT(x) x
+#else
+#define CONSTRAINT(x) typename
```

Comment:
Since we will likely have similar macros in the other projects as well, let's call this one `TRACCC_CONSTRAINT` instead. :thinking:

```suggestion
#define TRACCC_CONSTRAINT(x) x
#else
#define TRACCC_CONSTRAINT(x) typename
```

Reply:
Good point!